src/main/java/duke/CommandMatcher.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+/**
+ * This class serves as a way to abstract the idea of making a command
+ * as a matching process and an action.
+ */
+public class CommandMatcher {
+    private Predicate<String> shouldRunAction;
+    private Consumer<String> action;

```

Comment:
I like how you use the consumer functional class in your implementation, seems to be a rather unique way to do it